src/v3/verify.test.ts

```diff
@@ -0,0 +1,145 @@
+import { verify } from "./verify";
+import sample from "./schema/wrapped-sample-document.json";
+import batched from "./schema/wrapped-batched-documents-1.json";
+
+import { cloneDeep } from "lodash";
+import { OpenAttestationVerifiableCredential, SignatureAlgorithm } from "../shared/@types/document";
+
+// sample1: unwrapped (aka credential), sample2: only 1 doc is wrapped (aka verifiable credential/VC)
+const sampleVerifiableCredential = sample as OpenAttestationVerifiableCredential;
+
+// sample3 & sample4: more than 1 doc wrapped (aka batched VC, where 'proofs' has values)
+const sampleBatched1 = batched as OpenAttestationVerifiableCredential;
+
+describe("signature", () => {
+  describe("verify", () => {
+    test("returns true for documents with unaltered data (without proofs)", () => {
+      // For this test case, proofs does not exist as sampleVerifiableCredential is wrapped on its own (i.e. targetHash == merkleRoot)
+      const verifiableCredential = sampleVerifiableCredential;
+      const verified = verify(verifiableCredential);
+      expect(verified).toBe(true);
+    });
+    test("returns false for documents with altered value", () => {
+      const verifiableCredential = {
+        ...cloneDeep(sampleVerifiableCredential),
+        recipient: {
+          name: "Fake Name"
+        }
+      };
+      const verified = verify(verifiableCredential);
+      expect(verified).toBe(false);
+    });
+    test("returns false for documents with altered key", () => {
+      const verifiableCredential = {
+        ...cloneDeep(sampleVerifiableCredential),
+        recipient: {
+          fakename: "Recipient Name"
+        }
+      };
+      const verified = verify(verifiableCredential);
+      expect(verified).toBe(false);
+    });
+    test("returns false for documents with added data", () => {
```

Comment:
returns false for document with removed data ?

Reply:
fixed, changed to `returns false for documents with additional data not part of salt`